tfe/resource_tfe_policy_set_test.go

```diff
@@ -0,0 +1,617 @@
+			return fmt.Errorf("Bad value for global: %v", policySet.Global)
+		}
+
+		if len(policySet.Policies) != 1 {
+			return fmt.Errorf("Wrong number of policies: %v", len(policySet.Policies))
+		}
+
+		policyID := policySet.Policies[0].ID
+		policy, _ := tfeClient.Policies.Read(ctx, policyID)
+		if policy.Name != "policy-one" {
+			return fmt.Errorf("Wrong member policy: %v", policy.Name)
+		}
+
+		// Even though the terraform config should have 0 workspaces, the API will return
+		// workspaces for global policy sets. This list would be the same as listing the
+		// workspaces for the organization itself.
+		if len(policySet.Workspaces) != 1 {
+			return fmt.Errorf("Wrong number of workspaces: %v", len(policySet.Workspaces))
+		}
+
+		workspaceID := policySet.Workspaces[0].ID
+		workspace, _ := tfeClient.Workspaces.Read(ctx, "terraform-test", "workspace-one")
+		if workspace.ID != workspaceID {
+			return fmt.Errorf("Wrong member workspace: %v", workspace.Name)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckTFEPolicySetGlobalUpdate(policySet *tfe.PolicySet) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		tfeClient := testAccProvider.Meta().(*tfe.Client)
+
+		if policySet.Name != "populated-set" {
+			return fmt.Errorf("Bad name: %s", policySet.Name)
+		}
+
+		if policySet.Description != "A set populated with some policies" {
+			return fmt.Errorf("Bad description: %s", policySet.Description)
+		}
+
+		if policySet.Global {
+			return fmt.Errorf("Bad value for global: %v", policySet.Global)
+		}
+
+		if len(policySet.Policies) != 1 {
+			return fmt.Errorf("Wrong number of policies: %v", len(policySet.Policies))
+		}
+
+		policyID := policySet.Policies[0].ID
+		policy, _ := tfeClient.Policies.Read(ctx, policyID)
+		if policy.Name != "policy-one" {
+			return fmt.Errorf("Wrong member policy: %v", policy.Name)
+		}
+
+		if len(policySet.Workspaces) != 1 {
+			return fmt.Errorf("Wrong number of workspaces: %v", len(policySet.Workspaces))
+		}
+
+		workspaceID := policySet.Workspaces[0].ID
+		workspace, _ := tfeClient.Workspaces.Read(ctx, "terraform-test", "workspace-two")
+		if workspace.ID != workspaceID {
+			return fmt.Errorf("Wrong member workspace: %v", workspace.Name)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckTFEPolicySetDestroy(s *terraform.State) error {
+	tfeClient := testAccProvider.Meta().(*tfe.Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "tfe_policy_set" {
+			continue
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No instance ID is set")
+		}
+
+		_, err := tfeClient.PolicySets.Read(ctx, rs.Primary.ID)
+		if err == nil {
+			return fmt.Errorf("Sentinel policy %s still exists", rs.Primary.ID)
+		}
+	}
+
+	return nil
+}
+
+const testAccTFEPolicySet_basic = `
+resource "tfe_organization" "foobar" {
+  name = "terraform-test"
+  email = "admin@company.com"
+}
+
+resource "tfe_policy_set" "my_set" {
+  name = "my-set"
+	description = "A set for some of my Sentinel policies"
```

Comment:
Spaces/tabs? Or justy too many of either one of them :wink: Seems the same for all other `description` fields as well.